core/src/main/scala/com/nvidia/spark/rapids/tool/qualification/QualOutputWriter.scala

```diff
@@ -338,7 +339,7 @@ class QualOutputWriter(outputDir: String, reportReadSchema: Boolean,
     }
   }
 
-  def writeStatusReport(statusReports: Seq[StatusSummaryInfo], order: String): Unit = {
+  def writeStatusReport(statusReports: Seq[StatusProfileResult], order: String): Unit = {
```

Comment:
Should we make `StatusProfileResult` generic since its used by both tools?

Reply:
That sounds reasonable.
If moving that does not require too many changes, then will be good to do it in this PR.
Otherwise, we can keep it the way it is.

Reply:
I replaced `StatusProfileResult` to a more generic name `AppStatusResult`. `AppStatusResult` is defined in [ProfileClassWareHouse.scala](https://github.com/NVIDIA/spark-rapids-tools/blob/6fda70c4f2095cddece7446280c7c8d33434db0b/core/src/main/scala/com/nvidia/spark/rapids/tool/profiling/ProfileClassWarehouse.scala#L4). I think it is okay to keep the definition there because the file documentation says `This is a warehouse to store all Classes used for profiling and qualification`.